### PR TITLE
refactor: rename events to be consistent with WS names

### PR DESCRIPTION
**Pull request by kyranet**
**Please describe the changes this PR makes and why it should be merged:**

- `message` -> `messageCreate`
- `interaction` -> `interactionCreate`

| :warning: Old event names (`message` and `interaction`) will still work even after this PR, but a process warning will be emitted. |
| --- |

An alternative I thought of to emit the warning was to overload `Client#on`, `Client#once`, `Client#addListener`, 
`Client#prependListener`, and `Client#prependOnceListener`, which would have a much smaller performance impact, but it duplicates a lot of code.

**Status and versioning classification:**

- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by iCrawl — APPROVED** [review]

File: src/client/actions/MessageCreate.js [hunk 1]
```diff
@@ -25,10 +25,25 @@ class MessageCreateAction extends Action {
...
       client.emit(Events.MESSAGE_CREATE, message);
+      if (client.listenerCount('message')) {
+        process.emitWarning('message listener is deprecated', {
+          code: 'CLIENT_MESSAGE_EVENT',
+          detail: 'The event messageCreate should be used instead.',
+        });
+
+        /**
+         * Emitted whenever a message is created.
+         * @event Client#messageCreate
```
Comment (iCrawl) [inline src/client/actions/MessageCreate.js:40]:
```suggestion
         * @event Client#message
```

File: src/client/actions/InteractionCreate.js [hunk 2]
```diff
@@ -37,12 +39,27 @@ class InteractionCreateAction extends Action {
...
+
+    if (!deprecationEmitted && client.listenerCount('interaction')) {
+      deprecationEmitted = true;
+      process.emitWarning('The interaction event is deprecated. Use interactionCreate instead', 'DeprecationWarning');
+
+      /**
+       * Emitted when an interaction is created.
+       * @event Client#interaction
+       * @param {Interaction} interaction The interaction which was created
+       * @deprecated
```
Comment (vladfrangu) [inline src/client/actions/InteractionCreate.js:59]: Do we have "Use xyz instead" support here?

File: typings/index.d.ts [hunk 3]
```diff
@@ -3081,7 +3081,9 @@ declare module 'discord.js' {
     guildUpdate: [oldGuild: Guild, newGuild: Guild];
     inviteCreate: [invite: Invite];
     inviteDelete: [invite: Invite];
+    /** @deprecated */
```
Comment (vladfrangu) [inline typings/index.d.ts:3084]: Ditto

File: typings/index.d.ts [hunk 4]
```diff
@@ -3110,7 +3112,9 @@ declare module 'discord.js' {
     userUpdate: [oldUser: User | PartialUser, newUser: User];
     voiceStateUpdate: [oldState: VoiceState, newState: VoiceState];
     webhookUpdate: [channel: TextChannel];
+    /** @deprecated */
```
Comment (vladfrangu) [inline typings/index.d.ts:3115]: Ditto

**Review by vladfrangu — DISMISSED** [review]

File: src/client/actions/InteractionCreate.js (hunk 2 above)
Comment (kyranet) [reply src/client/actions/InteractionCreate.js:59]: Yes, we do: https://jsdoc.app/tags-deprecated.html

Comment (vladfrangu) [reply src/client/actions/InteractionCreate.js:59]: The question was about the website, not the JSDoc Tag

Comment (kyranet) [reply src/client/actions/InteractionCreate.js]: I have no idea, [MessageEmbed#type](https://discord.js.org/#/docs/main/master/class/MessageEmbed?scrollTo=type) is the only thing deprecated before this PR.

cc: @iCrawl

Comment (iCrawl) [reply src/client/actions/InteractionCreate.js]: We do not.

File: src/client/actions/InteractionCreate.js [hunk 5]
```diff
@@ -37,12 +39,27 @@ class InteractionCreateAction extends Action {
...
     /**
      * Emitted when an interaction is created.
-     * @event Client#interaction
+     * @event Client#interactionCreate
      * @param {Interaction} interaction The interaction which was created
      */
-    client.emit(Events.INTERACTION_CREATE, new InteractionType(client, data));
+    client.emit(Events.INTERACTION_CREATE, interaction);
+
+    if (!deprecationEmitted && client.listenerCount('interaction')) {
```
Comment (ckohen) [inline src/client/actions/InteractionCreate.js:51]: This prevents emitting the deprecated event after the process warning

File: src/client/actions/MessageCreate.js [hunk 6]
```diff
@@ -25,10 +27,23 @@ class MessageCreateAction extends Action {
 
       /**
        * Emitted whenever a message is created.
-       * @event Client#message
+       * @event Client#messageCreate
        * @param {Message} message The created message
        */
       client.emit(Events.MESSAGE_CREATE, message);
+      if (!deprecationEmitted && client.listenerCount('message')) {
```
Comment (ckohen) [inline src/client/actions/MessageCreate.js:34]: Ditto

File: typings/index.d.ts [hunk 7]
```diff
@@ -3081,7 +3081,9 @@ declare module 'discord.js' {
     guildUpdate: [oldGuild: Guild, newGuild: Guild];
     inviteCreate: [invite: Invite];
     inviteDelete: [invite: Invite];
+    /** @deprecated Use messageCreate instead */
```
Comment (SpaceEEC) [inline typings/index.d.ts:3084]: Are these visible anywhere except when looking at these lines in this file?

**Review by SpaceEEC — APPROVED** [review]

File: typings/index.d.ts (hunk 7 above)
Comment (kyranet) [reply typings/index.d.ts:3084]: They're only visible when you look up this property in the types, but doesn't show up when using `client.on('message', cb);` and others.

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**